pkg/v1/sdk/capabilities/discovery/README.md

```diff
@@ -8,58 +8,66 @@ The Query Builder SDK can query for:
 
 - Objects
   - Annotation
-  - Labels
-  - Conditions
 - Resources
   - WithFields
 - OpenAPI Schema
 
-Once created, these prepared queries can be exported and used whenever necessary .
+Once created, these prepared queries can be exported and used whenever necessary.
 
 ## Example
 
 ```go
-// Define some standard objects, GVRs or schema
-var ns = corev1.ObjectReference{
-    Kind:      "namespace",
-    Name:      "ian",
-    Namespace: "ian",
+import (
+    "sigs.k8s.io/controller-runtime/pkg/client/config"
+    "github.com/vmware-tanzu/tanzu-framework/pkg/v1/sdk/capabilities/discovery"
+)
+
+cfg := config.GetConfig()
+
+clusterQueryClient, err := discovery.NewClusterQueryClientForConfig(cfg)
+if err != nil {
+    log.Error(err)
 }
 
-// Its important to our use case for this annotation to also match
-var testAnnotations = map[string]string{
-    "cluster.x-k8s.io/provider": "infrastructure-fake",
+// Define objects to query.
+var pod = corev1.ObjectReference{
+    Kind:       "Pod",
+    Name:       "testpod",
+    Namespace:  "testns",
+    APIVersion: "v1",
 }
 
-// We want to ensure this version of pipelines exists, perhaps with specific fields
-var testGVR = schema.GroupVersionResource{
-    Group:    "tekton.dev",
-    Version:  "v1beta1",
-    Resource: "pipelines",
+var testAnnotations = map[string]string{
+    "cluster.x-k8s.io/provider": "infrastructure-fake",
 }
 
-// We will now generate a partial query to match our specific namespace
-var testResource1 = Object(ns).WithAnnotations(testAnnotations).WithConditions([C])
+// Define queries.
+var testObject = Object("podObj", &pod).WithAnnotations(testAnnotations)
 
-// Our usecase requires us to match a specific GVR
-var testGVR1 = GVR(testGVR).WithFields([field])
+var testGVR = Group("podResource", testapigroup.SchemeGroupVersion.Group).
+    WithVersions("v1").
+    WithResource("pods").
+    WithFields("spec.containers", "spec.serviceAccountName")
```

Comment:
nit: can we mention in the document that WithFields checks if field path exists only in the structural schema?